tensorflow/python/eager/context.py

```diff
@@ -1244,6 +1244,24 @@ def invoking_op_callbacks(self):
   def invoking_op_callbacks(self, value):
     self._thread_local_data.invoking_op_callbacks = value
 
+  # Parse PhysicalDevice from the device string
+  # if the device string contains subdevice type string, then parse it.
+  # if the device string doesn't contains subdevice type string, subdevice type will
+  # be the same as the device type string.
+  def _parse_physical_devices(self, devs):
+    physical_devices = []
+    for d in devs:
```

Comment:
Create a variable equal to `d.decode()` and use it below:
`device_spec = d.decode()`.

Also, I think it is better to rename input argument `devs` --> `devices` for clarity.

Reply:
Done